internal/services/api/post_inspect.go

```diff
@@ -59,126 +55,147 @@ func getEntrypointPath(projectDir util.AbsolutePath, w http.ResponseWriter, req
 	return relEntrypoint, nil
 }
 
-func PostInspectHandlerFunc(base util.AbsolutePath, log logging.Logger) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-		projectDir, relProjectDir, err := ProjectDirFromRequest(base, w, req, log)
-		if err != nil {
-			// Response already returned by ProjectDirFromRequest
-			return
-		}
-		dec := json.NewDecoder(req.Body)
-		dec.DisallowUnknownFields()
-		var b postInspectRequestBody
-		err = dec.Decode(&b)
-		if err != nil {
-			BadRequest(w, req, log, err)
-			return
-		}
-		pythonPath := util.NewPath(b.Python, nil)
-		rPath := util.NewPath(b.R, nil)
+type configGetter interface {
+	GetPossibleConfigs(base util.AbsolutePath, python util.Path, rExecutable util.Path, entrypoint util.RelativePath, log logging.Logger) ([]*config.Config, error)
+}
 
-		response := []postInspectResponseBody{}
+type postInspectHandler struct {
+	base                util.AbsolutePath
+	log                 logging.Logger
+	initializer         configGetter
+	matchingWalker      func([]string, util.AbsolutePath, logging.Logger) (util.Walker, error)
+	interpretersResolve func(util.AbsolutePath, http.ResponseWriter, *http.Request, logging.Logger) (interpreters.RInterpreter, interpreters.PythonInterpreter, error)
+}
```

Comment:
Sorry for the bump on changes, I had to re-wire a bit the inspect endpoint handler in order to be testable. We didn't had tests on this one so I went for it.

Reply:
No worries thanks for adding the additional tests!